packages/react-scripts/template/README.md

```diff
@@ -1408,12 +1375,157 @@ Example package.json:
+
+```json
+{
+  "scripts": {
+    "lint": "eslint src/"
+  }
+}
+```
+
+Then you can run the linter as follows:
+
+```bash
+npm run lint
+```
+
+> :warning: The ESLint rules that run with `npm start` and `npm build` **cannot** be customized or disabled. So if you make customizations as described above they won't run with `npm start` or `npm build`.
+
+### Failing ESLint on Warnings
+
+Most of the ESLint rules included with Create React App are configured as ESLint warnings and not errors. This means that a customized linter like the one described above often won't fail (exit with code 1) due to one of Create React App's ESLint rules. If you want ESLint to fail when it encounters warnings you can run ESLint with the `--max-warnings` flag set to 0:
+
+```json
+{
+  "scripts": {
+    "lint": "eslint src/ --max-warnings 0"
+  }
+}
+```
+
+### Integrating With Your Editor
+
+Some editors, including Sublime Text, Atom, and Visual Studio Code, provide plugins for ESLint.
+
+They are not required for linting. You should see the linter output right in your terminal as well as the browser console. However, if you prefer the lint results to appear right in your editor, there are some extra steps you can do.
+
+You would need to install an ESLint plugin for your editor first. Then, add a file called `.eslintrc` to the project root:
+
+```js
+{
+  "extends": "react-app"
+}
+```
+
+Now your editor should report the linting warnings.
+
+Note that even if you edit your `.eslintrc` file further, these changes will **only affect the editor integration**. They won’t affect the terminal and in-browser lint output. This is because Create React App intentionally provides a minimal set of rules that find common mistakes.
+
+If you want to enforce a coding style for your project, consider using [Prettier](https://github.com/jlongster/prettier) instead of ESLint style rules.
+
+### Using Global Variables
+
+When you include a script in the HTML file that defines global variables and try to use one of these variables in the code, the linter will complain because it cannot see the definition of the variable.
+
+You can avoid this by reading the global variable explicitly from the `window` object, for example:
+
+```js
+const $ = window.$;
+```
 
-When creating a build of your application with `npm run build` linter warnings are not checked by default. Like `npm test`, you can force the build to perform a linter warning check by setting the environment variable `CI`. If any warnings are encountered then the build fails.
+This makes it obvious you are using a global variable intentionally rather than because of a typo.
+
+Alternatively, you can force the linter to ignore any line by adding `// eslint-disable-line` after it.
+
+## Continuous Integration
+
+By default `npm test` runs the watcher with interactive CLI. However, you can force it to run tests once and finish the process by setting an environment variable called `CI`.
```

Comment:
Let's be consistent with above and say setting the `CI` environment variable to `true`.